lnwallet/channel.go

```diff
@@ -1790,6 +1790,30 @@ func (lc *LightningChannel) restoreStateLogs(
 		outgoingLocalAddHeights[l.HtlcIndex] = localCommitment.height
 	}
 
+	// If we have any unsigned acked updates to sign for, then the add is no
+	// longer on our local commitment, but is still on the remote's commitment.
+	// <---fail---
+	// <---sig----
+	// ----rev--->
+	// To ensure proper channel operation, we restore the add's addCommitHeightLocal
+	// field to the height of our local commitment.
+	for _, logUpdate := range unsignedAckedUpdates {
```

Comment:
Easier to follow now that it's all in once place, and also in the same area where we populate the map for the other HTLC cases.

Reply:
Yeah, nice. This is much easier to follow 👍